packages/headless/src/features/product-listing/product-listing-actions.ts

```diff
@@ -118,9 +118,11 @@ export const buildProductListingRequest = (
     ...(state.pagination && {
       pagination: {
         numberOfValues: state.pagination.numberOfResults,
-        page: Math.ceil(
-          state.pagination.firstResult / (state.pagination.numberOfResults || 1)
-        ),
+        page:
+          Math.ceil(
+            state.pagination.firstResult /
+              (state.pagination.numberOfResults || 1)
+          ) + 1,
```

Comment:
Is this calculation happening in the search use-case? If so, could we reuse the code in some way?

Reply:
I will happen, but since it's only used here, I'd suggest extracting it when we need it :thinking: